--- out/c-gcc4.6.s	2014-04-01 03:23:06.126775174 -0400
+++ out/c-gcc4.6-use-unreachable.s	2014-04-01 03:23:06.174775175 -0400
@@ -6,20 +6,7 @@
 _Z3fooj:
 .LFB0:
 	.cfi_startproc
-	cmpl	$100, %edi
 	movl	%edi, %eax
-	jbe	.L2
-	movl	%edi, %ecx
-	xorl	%edx, %edx
-	imull	%edi, %ecx
-	addl	$1, %ecx
-	imull	%edi, %ecx
-	imull	%ecx, %eax
-	addl	$1234, %ecx
-	addl	$1, %eax
-	divl	%ecx
-.L2:
-	rep
 	ret
 	.cfi_endproc
 .LFE0:
